homework1/task04.py

```diff
@@ -0,0 +1,31 @@
+"""
+Classic task, a kind of walnut for you
+
+Given four lists A, B, C, D of integer values,
+compute how many tuples (i, j, k, l) there are such that
+    A[i] + B[j] + C[k] + D[l] is zero.
+
+We guarantee, that all A, B, C, D have same length of N where 0 ≤ N ≤ 1000.
+"""
+from typing import List
+
+
+def check_sum_of_four(a: List[int], b: List[int],
+                      c: List[int], d: List[int]) -> int:
+    """
+    Using dict because it is a O(1) to check a values.
+    Similar to TwoSum from LeetCode
+    """
+    sums = {}
+    counter = 0
+    for i in a:
```

Comment:
Bad naming. i/j/k/l used as digits for enumeration, you have here an element of array.
for elem_a in a:
for elem_b in b... etc